### PR TITLE
Allow ipsec_mgmt_t read directories with ipsec_conf_file_t type BZ(17…

### PR DESCRIPTION
**Pull request by zpytela**
…73381)

### CODE REVIEW

**Comment by zpytela** [comment]
The fix is not complete yet.

**Comment by zpytela** [comment]
I decided to move the bug forward and propose a fix for both the parts. Please review, but consider the PR only for rawhide since the query on the strongswan maintainer is still in place.
```
# rpm -q selinux-policy; sesearch -A -s ipsec_mgmt_t -t ipsec_conf_file_t -c dir -p read; matchpathcon /usr/sbin/charon-systemd
selinux-policy-3.14.5-5.fc32.noarch
/usr/sbin/charon-systemd        system_u:object_r:bin_t:s0
# rpm -Fvh selinux-policy-*
Verifying...                          ################################# [100%]
Preparing...                          ################################# [100%]
Updating / installing...
   1:selinux-policy-3.14.5-16.200.bz17################################# [ 25%]
   2:selinux-policy-targeted-3.14.5-16################################# [ 50%]
Cleaning up / removing...
   3:selinux-policy-targeted-3.14.5-5.################################# [ 75%]
   4:selinux-policy-3.14.5-5.fc32     ################################# [100%]
# rpm -q selinux-policy; sesearch -A -s ipsec_mgmt_t -t ipsec_conf_file_t -c dir -p read; matchpathcon /usr/sbin/charon-systemd
selinux-policy-3.14.5-16.200.bz1773381.fc32.noarch
allow ipsec_mgmt_t ipsec_conf_file_t:dir read;
/usr/sbin/charon-systemd        system_u:object_r:ipsec_mgmt_exec_t:s0
```

File: policy/modules/system/ipsec.te [hunk 1]
```diff
@@ -291,6 +291,7 @@ allow ipsec_mgmt_t ipsec_t:unix_dgram_socket { create connect write };
 
 allow ipsec_mgmt_t ipsec_t:process { signal };
 
+allow ipsec_mgmt_t ipsec_conf_file_t:dir read;
```
Comment (wrabcak) [inline policy/modules/system/ipsec.te]: Why there is no permission set like "list_dir_perms" ?

**Review by wrabcak — CHANGES_REQUESTED** [review]

**Comment by mzabaluev** [comment]
Responded in https://bugzilla.redhat.com/show_bug.cgi?id=1773381
I agree with the proposed fix as far as I can understand SELinux policies. charon-systemd is a more modern binary to run strongswan as a native systemd service, so it needs the same context as the rest of the family. Don't know enough about the intricacies of directory permissions.

**Comment by wrabcak** [comment]
Thanks @zpytela for PR and @mzabaluev for review. Merging.

**Pull request merged**